libpod/image/pull.go

```diff
@@ -238,7 +239,17 @@ func (ir *Runtime) pullImageFromHeuristicSource(ctx context.Context, inputName s
 		if srcTransport != nil && srcTransport.Name() != DockerTransport {
 			return nil, err
 		}
+
+		fullnames, err := shortnames.Search(inputName)
```

Comment:
The integration should happen inside `ir.pullGoalFromPossiblyUnqualifiedName`, which already starts with a check whether the input specifies a registry (and in that case the short name lookup should not be involved).